python/ray/serve/tests/unit/test_router.py

```diff
@@ -489,5 +495,158 @@ async def test_max_queued_requests_updated(
         )
 
 
+def running_replica_info(replica_tag: str) -> RunningReplicaInfo:
+    return RunningReplicaInfo(
+        deployment_name="f",
+        replica_tag=replica_tag,
+        node_id="node_id",
+        availability_zone="some-az",
+        actor_handle=Mock(),
+        max_concurrent_queries=1,
+    )
+
+
+@patch("ray.serve._private.router.metrics.Counter", new=MockCounter)
+@patch("ray.serve._private.router.metrics.Gauge", new=MockGauge)
```

Comment:
Non-blocker: I think generally we prefer to do dependency injection rather than patching the objects. Just need to accept optional args for those objects and default to the real ones, so in the test you can pass in the mocked object. Might be a big change, can be in a followup.

Reply:
Yeah, I realized too that this is pretty annoying ... will follow up to discuss if we need to address this, if we make this change we should make it in other places as well (e.g. replica)